base/setup/reactos/drivepage.c

```diff
@@ -705,60 +1393,90 @@ DriveDlgProc(
         {
             LPNMHDR lpnm = (LPNMHDR)lParam;
 
-            // On Vista+ we can use TVN_ITEMCHANGED instead, with  NMTVITEMCHANGE* pointer
+            // On Vista+ we can use TVN_ITEMCHANGED instead, with NMTVITEMCHANGE* pointer
             if (lpnm->idFrom == IDC_PARTITION && lpnm->code == TVN_SELCHANGED)
             {
                 LPNMTREEVIEW pnmv = (LPNMTREEVIEW)lParam;
 
-                // if (pnmv->uChanged & TVIF_STATE) /* The state has changed */
-                if (pnmv->itemNew.mask & TVIF_STATE)
+                // if (!(pnmv->uChanged & TVIF_STATE)) /* The state has changed */
+                if (!(pnmv->itemNew.mask & TVIF_STATE))
+                    break;
+
+                /* The item has been (de)selected */
+                // if (pnmv->uNewState & TVIS_SELECTED)
+                if (pnmv->itemNew.state & TVIS_SELECTED)
                 {
-                    /* The item has been (de)selected */
-                    // if (pnmv->uNewState & TVIS_SELECTED)
-                    if (pnmv->itemNew.state & TVIS_SELECTED)
-                    {
-                        HTLITEM hParentItem = TreeList_GetParent(lpnm->hwndFrom, pnmv->itemNew.hItem);
-                        /* May or may not be a PPARTENTRY: this is a PPARTENTRY only when hParentItem != NULL */
-                        PPARTENTRY PartEntry = (PPARTENTRY)pnmv->itemNew.lParam;
+                    HTLITEM hParentItem = TreeList_GetParent(lpnm->hwndFrom, pnmv->itemNew.hItem);
+                    /* May or may not be a PPARTENTRY: this is a PPARTENTRY only when hParentItem != NULL */
 
-                        if (!hParentItem || !PartEntry)
+                    if (!hParentItem)
+                    {
+                        /* Hard disk */
+                        PDISKENTRY DiskEntry = (PDISKENTRY)pnmv->itemNew.lParam;
+                        ASSERT(DiskEntry);
+
+                        ShowWindow(GetDlgItem(hwndDlg, IDC_INITDISK), SW_SHOW);
+                        ShowWindow(GetDlgItem(hwndDlg, IDC_PARTCREATE), SW_HIDE);
+                        ShowWindow(GetDlgItem(hwndDlg, IDC_PARTDELETE), SW_HIDE);
+                    #if 0 // FIXME: Init disk not implemented yet!
+                        EnableDlgItem(hwndDlg, IDC_INITDISK,
+                                      DiskEntry->DiskStyle == PARTITION_STYLE_RAW);
+                    #else
+                        EnableDlgItem(hwndDlg, IDC_INITDISK, FALSE);
+                    #endif
```

Comment:
Move those preprocessor thingies to column 0
```suggestion
#if 0 // FIXME: Init disk not implemented yet!
                        EnableDlgItem(hwndDlg, IDC_INITDISK,
                                      DiskEntry->DiskStyle == PARTITION_STYLE_RAW);
#else
                        EnableDlgItem(hwndDlg, IDC_INITDISK, FALSE);
#endif
```